charts/opentelemetry-operator/release-checklist.md

```diff
@@ -0,0 +1,34 @@
+# Prerequisite
+
+- [ ] Make sure you have installed Kuttl which will be used to do the smoke tests later. See [Kuttl website](https://kuttl.dev/docs/)
+  for installation information.
+- [ ] Make sure you have cloned the [OpenTelemetry Operator](https://github.com/open-telemetry/opentelemetry-operator) in your workspace
+
+# Checklist
+
+- [ ] Change directory to `opentelemetry-helm-charts/charts/opentelemetry-operator`. `cd ./charts/opentelemetry-operator` should be helpful.
+- [ ] Run the command `go run main.go` to update the OTEL Collector CRD and get the latest tags of the two images which
+  the Operator will be using. Then update the corresponding values in the `values.yaml` (e.g., change the `manager.image.tag` value to `"v0.29.0"` or latest version).
+- [ ] Update `appVersion` value in the `Chart.yaml` if a new version of `manager.image.tag` is released
```

Comment:
Would be nice if the script can update values.yaml and Chart.yaml itself. Not asking to do anything here, maybe another follow up github issue?

Actually looks like all of the yaml -> helm templates steps from the checklist can be automated. At least validating up-to-dateness of the helm templates files.

Reply:
> Would be nice if the script can update values.yaml and Chart.yaml itself. Not asking to do anything here, maybe another follow up github issue?

Yes, we can update values.yaml just using the YAML marshal/unmarshal functions. But this will jeopardise the config option orders as well as the comments in the values.yaml. So, we decide to let the maintainers manually update them. I might be wrong and would be appreciated if there is a better way to do this!

> Actually looks like all of the yaml -> helm templates steps from the checklist can be automated. At least validating up-to-dateness of the helm templates files.

Yes, we can do this in another issue!

Reply:
> Yes, we can update values.yaml just using the YAML marshal/unmarshal functions. But this will jeopardise the config option orders as well as the comments in the values.yaml. So, we decide to let the maintainers manually update them. I might be wrong and would be appreciated if there is a better way to do this!

It can be done with grep/templating instead of yaml marshalling/unmarshalling, but not critical. At least we can check compare existing version in the values.yaml and Chart.yaml and ask maintainer to change only if it has to be updated. 

> Yes, we can do this in another issue!

Cool, please create an issue and we can close this convo

Reply:
I have rewritten the main.go to update the values.yaml and Chart.yaml using perl substitution function, and filed the [issue](https://github.com/open-telemetry/opentelemetry-helm-charts/issues/58) regarding the second concern. Thank you!